tests/test_end_to_end.py

```diff
@@ -122,6 +122,47 @@ def test_reference_netcdf(
         xr.testing.assert_equal(ds.load(), daily_xarray_dataset)
 
 
+def test_reference_netcdf_parallel(
+    daily_xarray_dataset,
+    netcdf_local_file_pattern_sequential_multivariable,
+    pipeline_parallel,
+    tmp_target,
+    output_file_name="reference.json",
+):
+    pattern = netcdf_local_file_pattern_sequential_multivariable
+    store_name = "daily-xarray-dataset"
+    with pipeline_parallel as p:
+        (
+            p
+            | beam.Create(pattern.items())
+            | OpenWithKerchunk(file_type=pattern.file_type)
+            | WriteCombinedReference(
+                identical_dims=["lat", "lon"],
+                target_root=tmp_target,
+                store_name=store_name,
+                concat_dims=["time"],
+                output_file_name=output_file_name,
+            )
+        )
+    full_path = os.path.join(tmp_target.root_path, store_name, output_file_name)
+
+    # Question: Should we add a test here for the number of keys in the reference
+    # or is that moot since the xarray dataset equality check is already in place?
+    # import json
+    # refs = json.loads(open(full_path).read())
+    # key_len = len(refs.keys())
+    # 25 keys expected for netcdf_local_paths_sequential_multivariable_1d:
+    # .zgroup + .zattrs + (3 dimensions * (.zattrs + .zarray + 1 chunk)) + \
+    #   (2 variables * (.zattrs + .zarray + 5 chunks))
+    # 35 keys expected for netcdf_local_paths_sequential_multivariable_2d
+    file_ext = os.path.splitext(output_file_name)[-1]
+
+    if file_ext == ".json":
+        mapper = fsspec.get_mapper("reference://", fo=full_path)
+        ds = xr.open_dataset(mapper, engine="zarr", backend_kwargs={"consolidated": False})
+        xr.testing.assert_equal(ds.load(), daily_xarray_dataset)
```

Comment:
This is expected to fail right now, until merged with npz/feature/simplify_zarr_merge

Reply:
Nice! Is there anyway we can change the target of this to be @moradology PR https://github.com/pangeo-forge/pangeo-forge-recipes/pull/689 and just merge it into there since he's trying to write tests already?

Reply:
Thanks @moradology let me know when that's ready. I've removed obsolete comments from this PR so it should be good to merge